url3426a97ee4_rtreffer_minidns_tgz-pJ8-de_measite_minidns_minidnsrepl_MiniDnsReplJ8/src/de/measite/minidns/dnsserverlookup/UnixUsingEtcResolvConf.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2015-2016 the original author or authors
+ *
+ * This software is licensed under the Apache License, Version 2.0,
+ * the GNU Lesser General Public License version 2 or later ("LGPL")
+ * and the WTFPL.
+ * You may choose either license to govern your use of this software only
+ * upon the condition that you accept all of the terms of either
+ * the Apache License 2.0, the LGPL 2.1+ or the WTFPL.
+ */
+package de.measite.minidns.dnsserverlookup;
+
+import de.measite.minidns.util.PlatformDetection;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class UnixUsingEtcResolvConf extends AbstractDNSServerLookupMechanism {
+
+    public static final DNSServerLookupMechanism INSTANCE = new UnixUsingEtcResolvConf();
+    public static final int PRIORITY = 2000;
+
+    private static final Logger LOGGER = Logger.getLogger(UnixUsingEtcResolvConf.class.getName());
+
+    private static final String RESOLV_CONF_FILE = "/etc/resolv.conf";
+    private static final Pattern NAMESERVER_PATTERN = Pattern.compile("^nameserver\\s+(.*)$");
+
+    private static String[] cached;
+    private static long lastModified;
+
+    private UnixUsingEtcResolvConf() {
+        super(UnixUsingEtcResolvConf.class.getSimpleName(), PRIORITY);
+    }
+
+    @Override
+    public String[] getDnsServerAddresses() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.